networkx/algorithms/centrality/incremental_closeness.py

```diff
@@ -0,0 +1,134 @@
+    prev_cc: list
+      The previous closeness centrality for all nodes in the graph.
+    insertion: bool, optional
+      If True (default) the edge was inserted, otherwise it was deleted from the graph.
+    normalized : bool, optional
+      If True (default) normalize by the number of nodes in the connected
+      part of the graph.
+
+    Returns
+    -------
+    nodes : dictionary
+      Dictionary of nodes with closeness centrality as the value.
+
+    See Also
+    --------
+    betweenness_centrality, load_centrality, eigenvector_centrality,
+    degree_centrality
+
+    Notes
+    -----
+    The closeness centrality is normalized to `(n-1)/(|G|-1)` where
+    `n` is the number of nodes in the connected part of graph
+    containing the node.  If the graph is not completely connected,
+    this algorithm computes the closeness centrality for each
+    connected part separately.
+
+    This algorithm is applicable to undirected graphs without edge weights.
+
+    References
+    ----------
+    .. [1] Freeman, L.C., 1979. Centrality in networks: I.
+       Conceptual clarification.  Social Networks 1, 215--239.
+       http://www.soc.ucsb.edu/faculty/friedkin/Syllabi/Soc146/Freeman78.PDF
+       [2]Sariyuce, A.E. ; Kaya, K. ; Saule, E. ; Catalyiirek, U.V. Incremental
+       Algorithms for Closeness Centrality
+       http://sariyuce.com/papers/bigdata13.pdf
+    """
+    if G.is_directed():
+        raise NetworkXNotImplemented("not implemented for directed graphs")
+
+    if prev_cc is not None and len(prev_cc) != len(G.nodes()):
+        raise NetworkXError(
+            "Previous closeness centrality list does not correspond to given\
+        graph.")
+
+    # Just aliases G
+    G_prime = G
+
+    # Unpack edge
+    (u, v) = edge
+    path_length = nx.single_source_shortest_path_length
+
+    if insertion:
+        # Do this first because G and G_prime refer to the same thing
+        du = path_length(G, u)
+        dv = path_length(G, v)
+
+        G_prime.add_edge(u, v)
+    else:
+        G_prime.remove_edge(u, v)
+
+        # For edge removal, we want to know about distances after the edge is gone
+        du = path_length(G_prime, u)
+        dv = path_length(G_prime, v)
+
+    if prev_cc is None:
+        return nx.closeness_centrality(G_prime)
+
+    nodes = G_prime.nodes()
+    closeness_centrality = {}
+    for n in nodes:
+        if abs(du[n] - dv[n]) <= 1:
+            closeness_centrality[n] = prev_cc[n]
+        else:
+            sp = path_length(G_prime, n)
+            totsp = sum(sp.values())
+            if totsp > 0.0 and len(G_prime) > 1:
+                closeness_centrality[n] = (len(sp) - 1.0) / totsp
+                # normalize to number of nodes-1 in connected part
+                if normalized:
+                    s = (len(sp) - 1.0) / (len(G_prime) - 1)
+                    closeness_centrality[n] *= s
+            else:
+                closeness_centrality[n] = 0.0
```

Comment:
So in some cases, the closeness value is zero. You could have a test case where such a value is set, as the statement doesn't seem to be covered by tests.